packages/web/src/components/nav/desktop/NavAudio.tsx

```diff
@@ -43,8 +71,12 @@ const NavAudio = () => {
   const { tier } = useSelectTierInfo(account?.user_id ?? 0)
   const audioBadge = audioTierMapPng[tier as BadgeTier]
 
-  const userChallenges = useSelector(getOptimisticUserChallenges)
-  const hasClaimableTokens = Object.values(userChallenges).some(
+  const optimisticUserChallenges = useSelector(getOptimisticUserChallenges)
+  const rewardIds = useRewardIds({})
+  const filteredUserChallenges = Object.values(
+    optimisticUserChallenges
+  ).filter(challenge => rewardIds.includes(challenge.challenge_id))
+  const hasClaimableTokens = filteredUserChallenges.some(
     challenge => challenge && challenge.claimableAmount > 0
```

Comment:
is challenge nullable here? guessing it shouldn't be after the filter?

Reply:
That seems right. And just checked getOptimisticUserChallenges, seems like it filters out nullable values before returning.